tncc.py

```diff
@@ -564,13 +571,34 @@ def get_cookie(self, dspreauth=None, dssignin=None):
         post_data = ''.join([ '%s=%s;' % (k, v) for k, v in post_attrs.iteritems()])
         self.r = self.br.open('https://' + self.vpn_host + self.path + 'hc/tnchcupdate.cgi', post_data)
 
+        # set the inital host checker cookie to the first DSPREAUTH cookie
+        self.set_cookie('DSPREAUTH_HC', self.find_cookie('DSPREAUTH').value)
+
+        # parse the response to retrieve the periodic host checking interval
+        response = self.parse_response()
+
+        # try to get the interval from the response, if not fallback with except
+        try:
+            self.hc_interval = int(response['msg'].split("interval=")[1].split("SESSION")[0])
+        except:
+            self.hc_interval = 10
+	
         # We have a new DSPREAUTH cookie
         return self.find_cookie('DSPREAUTH')
 
 class tncc_server(object):
     def __init__(self, s, t):
         self.sock = s
         self.tncc = t
+	self.hc_scheduler = sched.scheduler(time.time, time.sleep)
+
+
+    def do_hc(self, sc):
+        hc_cookie=self.tncc.find_cookie('DSPREAUTH_HC').value
+        self.tncc.setup_mechanize()
+	self.tncc.get_cookie(hc_cookie, 'url_default').value
+        logging.info("==== Next Host Checker scheduled...")
```

Comment:
Would not harm to include a number of seconds till next `do_hc` invocation in log output